### ISSUE TITLE
style function will ignore the baseline and vertically center the content in math mode

### ISSUE THREAD

**Issue by rqy2002** [typst/typst#602]
As in title. a minimal example is
```typst
#let f1 = box(baseline: 10pt, [f])
#let f2 = style(sty => f1)

f #f1 #f2

$f f1 f2$
```
in text mode `f #f1 #f2` works normally, but in mathmode `f2` will be higher than `f1`.

**Comment by rqy2002** [comment]
seems in formulas we use the generic `layout` function for `style`, which will wrap it with a `par`. but the layout of a `par` does not have a baseline.

maybe we should implement `layout_math` of all contents which have `Show` trait using `show` method.

But I'm a newbie rustacean and got trapped by rustc, it said I can't borrow something
(In fact, `ctx.vt` with `ctx.styles()`). Maybe we need to add a method to `MathContext`.

**Comment by laurmaedje** [comment]
Ideally, normal and math layout wouldn't be two completely separate worlds, but that will take some refactoring.

**Comment by laurmaedje** [comment]
Fixed via #637.
